argonaut/src/main/scala/org/http4s/argonaut/ArgonautInstances.scala

```diff
@@ -60,9 +63,37 @@ trait ArgonautInstances {
   }
 }
 
+sealed case class ArgonautInstancesBuilder private[argonaut] (
+    override val defaultPrettyParams: PrettyParams = PrettyParams.nospace,
+    override val jsonDecodeError: (Json, String, CursorHistory) => DecodeFailure =
+      ArgonautInstances.defaultJsonDecodeError,
+    override val jawnParseExceptionMessage: ParseException => DecodeFailure =
+      JawnInstances.defaultJawnParseExceptionMessage,
+    override val jawnEmptyBodyMessage: DecodeFailure = JawnInstances.defaultJawnEmptyBodyMessage
+) extends ArgonautInstances {
```

Comment:
It's a little strange for a "builder" to extend what it's building instead of have a terminal method that builds it, but I don't see the harm.  If there's no terminal `build` method, maybe these `with` methods should go straight onto `ArgonautInstances` and just make that the abstract class?

Reply:
I wanted to keep the `ArgonautInstances` a trait so it could still be used as a mixin.  I can definitely change the semantics of the builder to include a terminator so its less weird.